src/main/java/com/laytonsmith/core/compiler/analysis/StaticAnalysisConfiguration.java

```diff
@@ -0,0 +1,67 @@
+package com.laytonsmith.core.compiler.analysis;
+
+import com.laytonsmith.PureUtilities.Preferences;
+import com.laytonsmith.PureUtilities.Preferences.Preference;
+import com.laytonsmith.PureUtilities.Preferences.Type;
+import com.laytonsmith.abstraction.Implementation;
+import com.laytonsmith.core.MethodScriptFileLocations;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.logging.Logger;
+
+/**
+ *
+ */
+public final class StaticAnalysisConfiguration {
+
+	private static volatile StaticAnalysisConfiguration config = null;
+
+	public static StaticAnalysisConfiguration GetConfiguration() throws IOException {
+		@SuppressWarnings("LocalVariableHidesMemberVariable")
+		StaticAnalysisConfiguration config = StaticAnalysisConfiguration.config;
+		if(config == null) {
+			synchronized(StaticAnalysisConfiguration.class) {
+				config = StaticAnalysisConfiguration.config;
+				if(config == null) {
+					StaticAnalysisConfiguration.config = config = new StaticAnalysisConfiguration();
+					config.load();
+				}
+			}
+		}
+		return config;
+	}
+
+	private final Preferences prefs;
+
+	private StaticAnalysisConfiguration() {
+		List<Preferences.Preference> defaults = new ArrayList<>(Arrays.asList(
+			// TODO - Replace "off" with "on" as soon as static analysis should be enabled by default.
```

Comment:
Also need to add an upgrade routine to delete the config file, otherwise the previously set value will persist. But I'll remember that when this is changed, so don't need to change the TODO.